### PR TITLE
Refactored blog html and layouts

### PR DESCRIPTION
**Pull request by jehartzog**
One thing that took me a while to figure out was how the layouts were all structured to pull in and compress all the pages efficiently. The current layout is fantastic at putting everything together and compressing it, but I did find a few small file layout issues that didn't make sense to me.

This refactors that /blog/index.html to be structured the same as the rest of the root level files (about.md, tags.html, etc..), and cleans up the usage of includes in those files.

Basically, the end result is the same but the layout and code is more consistent and easy to follow.

### CODE REVIEW

**Comment by jehartzog** [comment]
Turns out `/blog/index.html` is needed for [pagination](https://jekyllrb.com/docs/pagination/), so I shifted that back. The other cleanup still applies.

**Comment by sergiokopplin** [comment]
Seems ok to me. No errors in my tests.

---

@jehartzog this PR is very helpful for the template.

---

Your PR makes me very happy and this whole thing about "open source" makes sense now.
